src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -0,0 +1,207 @@
+  
+  // No need to copy GCS inputs for the workflow we should be able to direclty reference them
+  override def initializeForWorkflow(workflow: WorkflowDescriptor): HostInputs = workflow.actualInputs
+
+  // FIXME: signature is weird
+  def localizeTaskOutput(taskOutput: TaskOutput, callGcsPath: String, scopedLookupFunction: ScopedLookupFunction, engineFunctions: JesEngineFunctions): Try[Option[JesOutput]] = {
+    taskOutput.wdlType match {
+      case WdlFileType =>
+        taskOutput.expression.evaluate(scopedLookupFunction, engineFunctions) match {
+          case Success(v) => Success(Option(JesOutput(taskOutput.name, s"$callGcsPath/${taskOutput.name}", Paths.get(v.valueString))))
+          case Failure(e) => Failure(new IllegalArgumentException(s"JES requires File outputs to be determined prior to running, but ${taskOutput.name} can not."))
+        }
+      case _ => Success(None)
+    }
+  }
+
+  // FIXME: Signature is weird
+  def localizeTaskOutputs(taskOutputs: Seq[TaskOutput], callGcsPath: String, scopedLookupFunction: ScopedLookupFunction, engineFunctions: JesEngineFunctions): Try[Seq[JesOutput]] = {
+    val localizedOutputs = taskOutputs map {localizeTaskOutput(_, callGcsPath, scopedLookupFunction, engineFunctions)}
+    val localizationFailures = localizedOutputs filter {_.isFailure}
+    if (localizationFailures.isEmpty) {
+      Success(localizedOutputs.collect({case Success(x) => x}).flatten)
+    } else {
+      val failureMessages = TryUtil.stringifyFailures(localizationFailures)
+      Failure(new IllegalArgumentException(failureMessages.mkString("\n")))
+    }
+  }
+
+  def anonymousTaskOutput(value: String, engineFunctions: JesEngineFunctions): JesOutput = {
+    JesOutput(value, engineFunctions.gcsPathFromAnyString(value).toString, Paths.get(value))
+  }
+
+  override def executeCommand(commandLine: String,
+                              workflowDescriptor: WorkflowDescriptor,
+                              call: Call,
+                              backendInputs: CallInputs,
+                              scopedLookupFunction: ScopedLookupFunction): Try[Map[String, WdlValue]] = {
+    val callGcsPath = s"${workflowDescriptor.callDir(call)}"
+
+    val engineFunctions = new JesEngineFunctions(GoogleCloudStoragePath(callGcsPath), JesConnection)
+
+    // FIXME: Not particularly robust at the moment.
+    val jesInputs: Seq[JesParameter] = backendInputs.collect({
+      case (k, v) if v.isInstanceOf[WdlFile] => JesInput(k, scopedLookupFunction(k).valueString, Paths.get(v.valueString))
+
+    }).toSeq
+
+    // Call preevaluateExpressionForFilenames for each of the task output expressions, and flatten the lists into a single Try[Seq[WdlFile]]
+    val filesWithinExpressions = Try(
+      call.task.outputs.map {
+        taskOutput => taskOutput.expression.preevaluateExpressionForFilenames(scopedLookupFunction, engineFunctions)
+      }.flatMap({_.get})
```

Comment:
I think we tried a for - kept getting stuck with incorrect (Try[None] I think) return types?